Scripts/Runtime/Entities/BufferFromSingleEntity.cs

```diff
@@ -0,0 +1,38 @@
+using Unity.Entities;
+
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// A container that provides access to a <see cref="DynamicBuffer{T}" from a single entitiy.
+    /// </summary>
+    /// <typeparam name="T">The element type of the buffer</typeparam>
+    /// <remarks>Allows developers to define jobs with fewer parameters that clearly communicate intent.</remarks>
+    public readonly struct BufferFromSingleEntity<T> where T : struct, IBufferElementData
```

Comment:
No idea if it matters but would this benefit from `[BurstCompile]`? 

For the `GetBuffer` function?

Reply:
Based on this thread and my tests it doesn't seem necessary.
https://forum.unity.com/threads/burst-question.772439/#post-5142527

I tried instantiating a managed type in the `GetBuffer` method and burst complained so the function call is getting burst compiled without the extra attribute.

Reply:
Ok cool, good to know